### PR TITLE
Change examples main

### PR DESCRIPTION
**Pull request by tohar-ayash**
Signed-off-by: tohar-ayash <toharayash1@gmail.com>

### CODE REVIEW

File: examples/api/acl_violation/acl_violation.go [hunk 1]
```diff
@@ -19,23 +20,29 @@ import (
 	each user opens a transaction that writes to the database, alice tx is valid and bob tx is non-valid
 */
 func main() {
+	if err := executeAclViolationExample(); err == nil {
+		os.Exit(1)
+	}
```
Comment (tock-ibm) [inline examples/api/acl_violation/acl_violation.go:25]: `err != nil` => `os.Exit(1)`

File: examples/api/db_tx/db_tx.go [hunk 2]
```diff
@@ -27,6 +35,10 @@ func main() {
 			Name:          "bcdb-client",
 		},
 	)
+	if err != nil {
+		fmt.Printf(err.Error())
+		return nil
```
Comment (tock-ibm) [inline examples/api/db_tx/db_tx.go:40]: return err

File: examples/api/tx_validity/tx_validity.go [hunk 3]
```diff
@@ -18,94 +19,65 @@ import (
 	In the case of async-commit, it's not possible to know in advance which one of the transactions will be valid, as the server may reorder them.
 */
 func main() {
-	c, err := util.ReadConfig("../../util/config.yml")
-	if err != nil {
-		fmt.Printf(err.Error())
+	if err := executeTxValidityExample(); err == nil {
```
Comment (tock-ibm) [inline examples/api/tx_validity/tx_validity.go:22]: err!=nil

File: examples/api/tx_validity/tx_validity.go [hunk 4]
```diff
@@ -262,22 +237,22 @@ LOOP:
 			}
 		case <-timeout:
 			fmt.Println("Getting transaction receipt failed")
-			return
+			return err
```
Comment (tock-ibm) [inline examples/api/tx_validity/tx_validity.go:240]: you have to create the error, i.e. errors.New("tx timeout")

File: examples/api/tx_validity/tx_validity.go [hunk 5]
```diff
@@ -262,22 +237,22 @@ LOOP:
...
 		}
 	}
 
 	tx3Flag := tx3Receipt.Header.ValidationInfo[tx3Receipt.TxIndex].Flag
 	fmt.Printf("Transaction number %s validation flag is %s\n", tx3Id, tx3Flag)
 	if tx3Flag != types.Flag_VALID {
 		fmt.Printf("Transaction number %s is invalid, reason: %s\n", tx3Id, tx3Receipt.Header.ValidationInfo[tx3Receipt.TxIndex].ReasonIfInvalid)
+	} else {
+		fmt.Printf("Transaction number %s committed successfully\n", tx3Id)
```
Comment (tock-ibm) [inline examples/api/tx_validity/tx_validity.go:249]: Which is the unexpected? return an error in that case to fail the test

File: examples/api/tx_validity/tx_validity.go [hunk 6]
```diff
@@ -297,22 +272,73 @@ LOOP2:
...
+			return err
 		}
 	}
 
 	tx4Flag := tx4Receipt.Header.ValidationInfo[tx4Receipt.TxIndex].Flag
 	fmt.Printf("Transaction number %s validation flag is %s\n", tx4Id, tx4Flag)
 	if tx4Flag != types.Flag_VALID {
 		fmt.Printf("Transaction number %s is invalid, reason: %s\n", tx4Id, tx4Receipt.Header.ValidationInfo[tx4Receipt.TxIndex].ReasonIfInvalid)
+	} else {
+		fmt.Printf("Transaction number %s committed successfully\n", tx4Id)
```
Comment (tock-ibm) [inline examples/api/tx_validity/tx_validity.go:284]: Which is the unexpected? return an error in that case to fail the test

Comment (tock-ibm) [reply examples/api/tx_validity/tx_validity.go:284]: or do it below in lines 287-292

File: examples/api/acl_violation/acl_violation.go (hunk 1 above)
Comment (tohar-ayash) [reply examples/api/acl_violation/acl_violation.go:25]: Done

File: examples/api/db_tx/db_tx.go (hunk 2 above)
Comment (tohar-ayash) [reply examples/api/db_tx/db_tx.go:40]: Done

File: examples/api/tx_validity/tx_validity.go (hunk 3 above)
Comment (tohar-ayash) [reply examples/api/tx_validity/tx_validity.go:22]: Done

File: examples/api/tx_validity/tx_validity.go (hunk 4 above)
Comment (tohar-ayash) [reply examples/api/tx_validity/tx_validity.go:240]: Done

File: examples/api/tx_validity/tx_validity.go [hunk 7]
```diff
@@ -297,22 +272,73 @@ LOOP2:
...
 	}
 
-	fmt.Printf("Transaction number %s committed successfully\n", tx4Id)
-
 	if tx3Flag == types.Flag_VALID && tx4Flag == types.Flag_VALID {
 		println("Error - both tx3 and tx4 are valid")
 	}
 	if tx3Flag != types.Flag_VALID && tx4Flag != types.Flag_VALID {
 		println("Error - both tx3 and tx4 are invalid")
 	}
```
Comment (tock-ibm) [inline examples/api/tx_validity/tx_validity.go:292]: instead of printing an error, also exit with an error

Comment (tohar-ayash) [reply examples/api/tx_validity/tx_validity.go]: Done

**Review by tock-ibm — APPROVED** [review]

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**